demo_k6_rp_agent.js

```diff
@@ -1,47 +1,60 @@
-var testConfig = JSON.parse(open(configFilePath));
-var reporterOptions = testConfig.reporterOptions;
+const reporterOptions = JSON.parse(open(__ENV.TEST_CONFIG)).reporterOptions;
 
 export const options = {
     scenarios: {
-        "my-sceanrio": {
-            executor: "shared-iterations",
+        'my-sceanrio': {
+            executor: 'shared-iterations',
             vus: 1,
             iterations: 3,
-            maxDuration: "10m",
+            maxDuration: '10m',
         },
     },
 };
 
-export function setup()
-{
-
-    let launchId = startLaunch(reporterOptions);
+export function setup() {
+    const launchId = startLaunch(reporterOptions);
     console.log(launchId);
-    return {"launchId": launchId}
+    return { 'launchId': launchId }
 }
-export default function(data)
-{
+export default function (data) {
+    let logBatch = []; // Defining array for log batch to be uploaded later
     const execIteration = exec.scenario.iterationInTest;
-    let launchId = data.launchId
-    let rpClient = new RpClient(launchId,reporterOptions);
-    let suiteId =rpClient.startSuite(`(Suite) Services #${execIteration+1}`,"Performance Test Workflow");
+    const launchId = data.launchId
+    const rpClient = new RpClient(launchId, reporterOptions);
+    const suiteId = rpClient.startSuite(`(Suite) Services #${execIteration + 1}`, 'Performance Test Workflow');
     console.log(suiteId);
-    let testId =rpClient.startTest(suiteId,"(Test) PluginServiceTest","Test Plugin service");
+    const testId = rpClient.startTest(suiteId, '(Test) PluginServiceTest', 'Test Plugin service');
     console.log(testId);
-    let uploadPlugintestStepId =rpClient.startTestStep(testId,"(Step) uploadPlugin","(Step) uploadPlugin");
+    const uploadPlugintestStepId = rpClient.startTestStep(testId, '(Step) Check k6 get', '(Step) checkK6get');
+    const res = http.get('http://test.k6.io');
+    if (
+        !check(res, {
+            'status code MUST be 200': (response) => response.status == 200,
+        })
+    ) {
+        rpClient.saveSingleLog('Response was not 200', 'error'); // Save single log
+        fail('status code was *not* 200');
+    }
+    logBatch = rpClient.addLogToBatch(logBatch, 'Response *was* 200', 'info'); // Add first log to batch
     console.log(uploadPlugintestStepId);
-    let updatePlugintestStepId =rpClient.startTestStep(testId,"(Step) updatePlugin","(Step) updatePlugin");
-    //finishTestStep(testStepId,"passed");
-    rpClient.finishTestStep(uploadPlugintestStepId,"failed","ab001","(Step) uploadPlugin Failed");
-    rpClient.finishTestStep(updatePlugintestStepId,"passed"); // As passed , it doesn't have issue type and comment
-    rpClient.finishTest(testId,"failed");
+    const updatePlugintestStepId = rpClient.startTestStep(testId, '(Step) Sleep 1 second', '(Step) Sleep');
+    sleep(1);
+    rpClient.saveSingleLog('Sleep', 'info');
+    logBatch = rpClient.addLogToBatch(logBatch, 'Sleep', 'info'); // Add second log to batch
```

Comment:
logBatch is passed by reference to method ,so it will be appended already. Why again returning and overriding in logBatch variable?

Reply:
If you check the function itself, you will see that I do not change the value of the function argument, I assign it to the function's local variable and then return the renewed (with new logs added) value from the function using that same local variable. This is why I overwrite an existing variable in executable file's default function using a callback. 
This practise ensures that I do not breach the scope of agent's function, as changing the value of passed argument would be a bad practise from the OOP standpoint. 
More on that topic can be found [here](https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern).
Nonetheless, this function's usage is optional, you can generate a log batch by any means comfortable to you within the default function or by using any third party file with a defined log aggregator.